sui-execution/latest/sui-adapter/src/adapter.rs

```diff
@@ -204,6 +183,42 @@ mod checked {
                 ])
                 .inc();
         }
+
+        Ok(())
+    }
+
+    /// Run both the Move verifier and the Sui verifier, checking just for timeouts. Returns Ok(())
+    /// if the verifier completes within the module meter limit and the ticks are successfully
+    /// transfered to the package limit (regardless of whether verification succeeds or not).
+    fn verify_module_timeout_only(
+        module: &CompiledModule,
+        verifier_config: &VerifierConfig,
+        meter: &mut (impl Meter + ?Sized),
+    ) -> Result<(), SuiError> {
+        meter.enter_scope(module.self_id().name().as_str(), Scope::Module);
+
+        if let Err(e) = verify_module_with_config_metered(verifier_config, module, meter) {
+            // Check that the status indicates metering timeout.
+            if check_for_verifier_timeout(&e.major_status()) {
+                return Err(SuiError::ModuleVerificationFailure {
+                    error: format!("Verification timed out: {}", e),
+                });
+            }
+        } else if let Err(err) = sui_verify_module_metered_check_timeout_only(
+            module,
+            &BTreeMap::new(),
+            meter,
+            verifier_config,
+        ) {
+            return Err(err.into());
+        }
+
+        if meter.transfer(Scope::Module, Scope::Package, 1.0).is_err() {
```

Comment:
This is the spot where we might do more work, right? 

And to check my understanding, the amount of additional work is bounded by the amount of work permissible for a given module?

Reply:
That's right, so we may spend as much as 2x the current tick limit before we realise we've expended the limit with this change. I think we could potentially be cleverer with the meter implementation though, to have the package donate its limit (up to the module limit) to the module when you enter it, and have the module return the unused limit when it comes back.

Reply:
This ...didn't work for a couple of reasons:

- Originally, I tried what I mentioned above, but it gets very messy because we don't keep track of where we are entering a scope from (nothing in the interface preventing us from entering a package from within a module, etc), so there isn't a good way to know which scope is donating to the one being entered.
- Then I tried keeping a global pool of ticks that every scope pulls from (and donates back to on `transfer`). This also doesn't work, because of the scale factor parameter on `transfer`.

TL;DR: The `Meter` trait confounds me once again.